enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,216 @@
+---
+title: microshift-csi-snapshot-integration
+authors:
+  - @copejon
+reviewers:
+  - @dhellman
+  - @pmtk
+  - @eggfoobar
+  - @pacevedom
+approvers:
+  - @dhellmann
+api-approvers:
+  - None
+creation-date: 2023-05-10
+last-updated: 2023-05-25
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-1140
+see-also: []
+---
+
+# CSI Snapshotting Integration
+
+## Summary
+
+MicroShift is a small form-factor, single-node OpenShift targeting IoT and Edge Computing use cases characterized by tight resource constraints, unpredictable network connectivity, and single-tenant workloads. See [kubernetes-for-devices-edge.md](./kubernetes-for-device-edge.md) for more detail.
+
+This document proposes the integration of the CSI Snapshot Controller to support backup and restore scenarios for cluster workloads.  The snapshot controller, along with the CSI external snapshot sidecar, will provide an API driven pattern for managing stateful workload data.
+
+
+## Motivation
+
+CSI snapshot functionality was originally excluded from the CSI driver integration in MicroShift to support the low-resource overhead goals of the project. However, user feedback has made it clear that a supportable, robust backup/restore solution is necessary.  While it would be possible to run a workflow out-of-band to manage workload data, this would be reinventing the wheel and contribute significantly to technical debt.  CSI Snapshots are already integrated into Openshift with ongoing downstream support. 
+
+### User Stories
+
+As an edge device owner running stateful workloads on MicroShift, I want to create in-cluster snapshots of that state and to restore workloads to that state utilizing existing Kubernetes patterns.
+
+### Goals
+
+* Enable an in-cluster workflow for snapshotting and restoration of cluster workload data
+* Follow the [MicroShift design principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* Enable backup/restore workflows outside of MicroShift
+* Provide a workflow for exporting data outside of a MicroShift cluster
```

Comment:
I think we also said we wouldn't worry about making this optional, for now, that we would address that holistically in another design.